src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -489,6 +489,10 @@ def __init__(self, **kwargs):
         # Get the list of I/O options that user has passed in
         ioarg_names = [name for name in kwargs.keys() if name not in ['data', 'link_data', 'allow_plugin_filters',
                                                                       'dtype', 'shape']]
+
+        if kwargs['data'] is None:
+            if kwargs['shape'] is None or kwargs['dtype'] is None:
+                raise ValueError("Must specify 'dtype' and 'shape' if not specifying 'data'")
```

Comment:
Should this check be part of ``DataIO.__init__`` instead? I can't think of a use case where we want to allow this in ``DataIO`` (or any subclass thereof) but not in ``HDF5IO``.

Reply:
Yes, it has been moved to DataIO